docs/user/implement_algo.rst

```diff
@@ -348,9 +358,107 @@ For completeness, the full experiment file is repeated below:
+                                                         name='return')
+            policy_dist = self.policy.build(self._obs, name='policy').dist
+            with tf.name_scope('loss'):
+                ll = policy_dist.log_prob(self._action, name='log_likelihood')
+                loss = -tf.reduce_mean(ll * self._returns)
+            with tf.name_scope('train'):
+                self._train_op = tf.compat.v1.train.AdamOptimizer(
+                    self._lr).minimize(loss)
+
+        def train(self, runner):
+            for epoch in runner.step_epochs():
+                samples = runner.obtain_samples(epoch)
+                log_performance(epoch,
+                                TrajectoryBatch.from_trajectory_list(self.env_spec,
+                                                                     samples),
+                                self._discount)
+                self.train_once(samples)
+
+        def train_once(self, paths):
+            obs = [path['observations'] for path in paths]
+            obs = tensor_utils.pad_tensor_n(obs, self.max_path_length)
+            actions = [path['actions'] for path in paths]
+            actions = tensor_utils.pad_tensor_n(actions, self.max_path_length)
+            returns = []
+            for path in paths:
+                returns.append(tensor_utils.discount_cumsum(path['rewards'],
+                                                           self._discount))
+            returns = tensor_utils.pad_tensor_n(returns, self.max_path_length)
+            sess = tf.compat.v1.get_default_session()
+            sess.run(self._train_op, feed_dict={
+                 self._obs: np.stack(obs, axis=0),
+                 self._action: np.stack(actions, axis=0),
+                 self._returns: np.stack(returns, axis=0),
+            })
+            return np.mean(returns)
+
+        # For pickling
+        def __getstate__(self):
+            data = self.__dict__.copy()
+            del data['_obs']
+            del data['_action']
+            del data['_train_op']
+            del data['_returns']
+            return data
+
+    from garage import wrap_experiment
+    from garage.envs import PointEnv, GarageEnv
+    from garage.experiment import LocalTFRunner
+    from garage.experiment.deterministic import set_seed
+    from garage.tf.policies import GaussianMLPPolicy
+
+    @wrap_experiment(log_dir='my_algorithm_logs', use_existing_dir=True, archive_launch_repo=False)
+    def debug_my_algorithm(ctxt):
+        set_seed(100)
+        with LocalTFRunner(ctxt) as runner:
+            env = GarageEnv(PointEnv())
+            policy = GaussianMLPPolicy(env.spec)
+            algo = MyAlgorithm(env.spec, policy)
+            runner.setup(algo, env)
+            runner.train(n_epochs=500, batch_size=4000)
+
+    debug_my_algorithm()
+
 
 .. [1] Williams, Ronald J. "Simple statistical gradient-following algorithms for connectionist reinforcement learning." Machine learning 8.3-4 (1992): 229-256.
```

Comment:
please use bibtex for generating these citations. See https://github.com/rlworkgroup/garage/pull/1765 for an example.